srcpkgs/efibootmgr/files/kernel.d/efibootmgr.post-install

```diff
@@ -28,18 +28,25 @@ major_version=$(echo $PKGNAME | cut -c 6-)
 # look for previous entry for this major kernel version
 existing_entry=$(efibootmgr | grep "Void Linux with kernel ${major_version}")
 
-# get the boot order
-# this is required because when in the next step the existing entry is removed,
-# it is also removed from the order so it needs to be restored later
-bootorder=$(efibootmgr |grep "BootOrder: " |cut -c 12-)
-
 # if existing, remove it
 if [ "$existing_entry" != "" ]; then
-  /etc/kernel.d/post-remove/50-efibootmgr $PKGNAME
+	/etc/kernel.d/post-remove/50-efibootmgr $PKGNAME
 fi
 
-# create the new entry
-efibootmgr -qc $args -L "Void Linux with kernel ${major_version}" -l /vmlinuz-${VERSION} -u "${OPTIONS}"
-
-# restore the boot order
-efibootmgr -qo $bootorder
```

Comment:
Won't removing this mean I might end up booting a different kernel than the one I want?

Reply:
BTW, there's another bug wrt bootorder: if the bootorder list has an empty slot, saving the order and then restoring like this script will change the entries after the empty slot.

Reply:
> BTW, there's another bug wrt bootorder: if the bootorder list has an empty slot, saving the order and then restoring like this script will change the entries after the empty slot.

Sorry, my explanation is a mess. What happens is:
 - say kernel is on entry 0005, which is the first in bootorder
 - now install a new version of the kernel
 - script removes entry for this kernel (0005)
 - script adds entry for the new version of the kernel
Result: if 0000-0004 are occupied then the new kernel will be 0005, but if one of 0000-0004 is empty, new kernel will go to the empty slot and 0005 will be empty. Then the bootorder is screwed.

The better way to do it would be:
 - script checks if there's an entry for the kernel it's replacing
 - if it is, it removes this old entry, and uses the same number for the new entry

Reply:
I've encountered this before, I thought it was because of `efibootmgr -qo $bootorder` so I removed it. What exactly do you want right now?

Reply:
If not fixing the issue I described above, I would say at least leave the `efibootmgr -qo $bootorder` in place. Without it, the last kernel installed will take priority which is not very convenient (e.g. I have 4.19, 5.4, 5.10 and 5.13 installed, and I'm currently booting 5.4; I would hate that this changes every time one of the others is updated).

A possible fix might be to do something like this:
```
[...]
# get major version, e.g. "4.8" for "linux4.8"
major_version=$(echo $PKGNAME | cut -c 6-)

label="Void Linux with kernel ${major_version}"

# look for previous entry for this major kernel version
bootnum=$(efibootmgr | grep " $label$" | cut -c "5-8")

# get the boot order
# this is required because when in the next step the existing entry is removed,
# it is also removed from the order so it needs to be restored later
bootorder=$(efibootmgr | grep "^BootOrder: " |cut -c 12-)

# if existing, remove it and reuse the same number
if [ -n "$bootnum" ]; then
	args="$args -B -b $bootnum"
fi

# if there is a boot order, keep it as is
if [ -n "$bootorder" ]; then
	args="$args -o $bootorder"
fi

# create the new entry
efibootmgr -qc $args -L "$label" -l /vmlinuz-${VERSION} -u "${OPTIONS}"
```

Instead of calling the `post-remove/50-efibootmgr` hook, just tell efibootmgr to replace the entry. This also keeps the order in the same call to efibootmgr so you don't have to do that in a separate step anymore.

Reply:
I also added the following to the end of the kernel hook script, which is purely informational and useful:
```
##### Give some feedback

# reread $bootnum and $bootorder after the change
bootnum=$(efibootmgr | grep " $label$" | cut -c "5-8")
bootorder=$(efibootmgr | grep "^BootOrder: " |cut -c 12-)
default=$(efibootmgr | grep "^Boot${bootorder%%,*}" | cut -c 11-)

if [ -z "${bootorder##$bootnum*}" ]; then
	# $bootnum is the first entry in $bootorder
	echo "  Linux kernel ${VERSION} will boot by default"
else
	# new order with $bootnum first
	neworder=$bootnum,$(echo $bootorder | sed -e "s/,$bootnum//")
	echo "  Linux kernel ${VERSION} in boot entry '$bootnum' will NOT boot by default"
	echo "  Current default is '$default'"
	#echo "  Current boot order is $bootorder"
	#( IFS="," ; for num in $bootorder ; do 
	#	echo -n "    "
	#	efibootmgr | grep "^Boot$num"
	#done )
	echo "   - run 'efibootmgr -o $neworder' to boot kernel ${VERSION} by default"
	echo "   - run 'efibootmgr -n $bootnum' to boot it next time only"
fi
```

Reply:
These are great ideas, but they shouldn't be part of this pr. A new pr for these changes would be lovely, but not this pr.

Reply:
> These are great ideas, but they shouldn't be part of this pr. A new pr for these changes would be lovely, but not this pr.

Then move 
```diff
-# restore the boot order
-efibootmgr -qo $bootorder
```
to a different issue/PR.